### PR TITLE
[ASR] convert not(x) into IntegerBitNot (fix #800)

### LINKED ISSUE #800: ASR should convert `not(x)` to `IntegerBitNot` node

### PR DESCRIPTION
**Pull request by lucifer1004**

### CODE REVIEW

**Comment by lucifer1004** [comment]
Can anyone help me to regenerate the LLVM reference to fix the CI? I cannot do it locally because my LLVM does not emit `align = xx` information.

File: src/lfortran/semantics/ast_common_visitor.h [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #ifndef LFORTRAN_SEMANTICS_AST_COMMON_VISITOR_H
 #define LFORTRAN_SEMANTICS_AST_COMMON_VISITOR_H
 
+#include "libasr/assert.h"
```
Comment (certik) [inline src/lfortran/semantics/ast_common_visitor.h:4]:
```suggestion
#include <libasr/assert.h>
```

**Comment by certik** [comment]
I pushed in a commit with the updated tests.

I noticed different LLVM versions emit slightly different LLVM IR in textual form, so unfortunately the tests depend on the LLVM version.

**Comment by lucifer1004** [comment]
@certik Could you please review this?

**Review by certik — APPROVED** [review]
I think this looks good.

**Comment by certik** [comment]
Thanks for the fix!

**Pull request merged**